File: static/js/guess.js
// 为开始游戏按钮添加点击事件监听器
document.getElementById('start-button').addEventListener('click', function() {
    // 向服务器发送请求以启动游戏
    fetch('/start_guess', {
        method: 'POST' // 使用 POST 方法
    })
    .then(response => response.json()) // 解析返回的 JSON 响应
    .then(data => {
        document.getElementById('result').innerText = data.status; // 显示游戏状态
        document.getElementById('guess-input').disabled = false; // 启用输入框
        document.getElementById('guess-button').disabled = false; // 启用猜测按钮
        document.getElementById('guess-input').value = ''; // 清空输入框
    })
    .catch(error => {
        console.error('Error:', error); // 处理错误
    });
});

// 为猜测按钮添加点击事件监听器
document.getElementById('guess-button').addEventListener('click', function() {
    const guess = parseInt(document.getElementById('guess-input').value); // 获取并解析用户输入的猜测
    // 向服务器发送猜测请求
    fetch('/guess', {
        method: 'POST', // 使用 POST 方法
        headers: {
            'Content-Type': 'application/json' // 设置请求头为 JSON 格式
        },
        body: JSON.stringify({ number: guess }) // 将猜测数字转为 JSON 字符串
    })
    .then(response => response.json()) // 解析返回的 JSON 响应
    .then(data => {
        document.getElementById('result').innerText = data.message; // 显示结果信息
        if (data.attempts !== undefined) { // 如果有尝试次数信息
            document.getElementById('result').innerText += ` (第 ${data.attempts} 次猜测)`; // 显示猜测次数
        }
        if (data.correct_number !== undefined) { // 如果猜对了
            document.getElementById('guess-input').disabled = true; // 禁用输入框
            document.getElementById('guess-button').disabled = true; // 禁用猜测按钮
        }
    })
    .catch(error => {
        console.error('Error:', error); // 处理错误
    });
});

// 添加重复的事件监听器（可以去掉）
document.getElementById('start-button').addEventListener('click', startGame);
document.getElementById('guess-button').addEventListener('click', makeGuess);

// 启动游戏的函数
function startGame() {
    // 向服务器发送请求以启动游戏
    fetch('/guess/start_guess', {
        method: 'POST', // 使用 POST 方法
        headers: {
            'Content-Type': 'application/json' // 设置请求头为 JSON 格式
        }
    })
    .then(response => response.json()) // 解析返回的 JSON 响应
    .then(data => {
        if (data.status === "Game started") { // 检查游戏是否成功启动
            document.getElementById('result').innerText = "游戏已开始！请猜一个数字。"; // 显示游戏开始信息
            document.getElementById('guess-input').disabled = false; // 启用输入框
            document.getElementById('guess-button').disabled = false; // 启用猜测按钮
        }
    })
    .catch(error => console.error('Error:', error)); // 处理错误
}

// 处理用户猜测的函数
function makeGuess() {
    const userGuess = parseInt(document.getElementById('guess-input').value, 10); // 获取并解析用户输入的猜测

    // 向服务器发送猜测请求
    fetch('/guess/guess', {
        method: 'POST', // 使用 POST 方法
        headers: {
            'Content-Type': 'application/json' // 设置请求头为 JSON 格式
        },
        body: JSON.stringify({ number: userGuess }) // 将猜测数字转为 JSON 字符串
    })
    .then(response => response.json()) // 解析返回的 JSON 响应
    .then(data => {
        if (data.error) { // 如果返回了错误信息
            document.getElementById('result').innerText = data.error; // 显示错误信息
        } else {
            const status = data.status || "unknown"; // 获取状态，默认值为 "unknown"
            document.getElementById('result').innerText = `你的猜测是 ${userGuess}，结果是：${data.message}。`; // 显示猜测结果
            if (data.remaining !== undefined) { // 如果有剩余猜测次数信息
                document.getElementById('result').innerText += ` 剩余猜测次数：${data.remaining}`; // 显示剩余次数
            }
            if (status === "correct") { // 如果猜对了
                document.getElementById('guess-input').disabled = true; // 禁用输入框
                document.getElementById('guess-button').disabled = true; // 禁用猜测按钮
            }
        }
    })
    .catch(error => console.error('Error:', error)); // 处理错误
}

File: logic/guess.py
import random  # 导入随机模块用于生成随机数

class Guess:
    def __init__(self):
        # 初始化游戏状态
        self.target_number = None  # 目标数字
        self.guesses = []  # 存储用户的猜测
        self.max_guesses = 10  # 最大猜测次数
        self.attempts_left = self.max_guesses  # 剩余猜测次数

    def start(self):
        # 启动游戏
        self.target_number = random.randint(0, 100)  # 随机生成一个 0 到 100 之间的目标数字
        self.guesses = []  # 清空之前的猜测
        self.attempts_left = self.max_guesses  # 重置剩余猜测次数
        return {"message": "游戏已开始，请猜一个数字。", "max_guesses": self.max_guesses}  # 返回游戏启动信息

    def make_guess(self, guess):
        # 处理用户的猜测
        if self.target_number is None:
            return {"status": "error", "message": "游戏未开始，请先启动游戏。"}  # 检查游戏是否已启动

        if self.attempts_left <= 0:
            return {
                "status": "game_over",  # 发送游戏结束状态
                "message": f"很遗憾，正确答案是：{self.target_number}",  # 游戏结束时返回正确答案
                "correct_number": self.target_number
            }

        self.guesses.append(guess)  # 记录用户的猜测
        self.attempts_left -= 1  # 每次猜测后减少剩余尝试次数

        # 判断猜测结果
        if guess < self.target_number:
            return {"status": "too_low", "message": "太小了", "attempts": len(self.guesses), "remaining": self.attempts_left}  # 猜的数字太小
        elif guess > self.target_number:
            return {"status": "too_high", "message": "太大了", "attempts": len(self.guesses), "remaining": self.attempts_left}  # 猜的数字太大
        else:
            return {
                "status": "correct",
                "message": f"正确答案就是：{self.target_number}",  # 猜对了
                "attempts": len(self.guesses),
                "correct": True
            }